airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
+        if usage:
+            view.usage.fill(usage)
+        if release_ver:
+            view.release_version.fill(release_ver)
+        if add_ons:
+            view.add_ons.fill(add_ons)
+        view.save.click()
+
+    def add_host_to_host_collection(
+        self, entity_name, host_collection_name=None, add_to_all_collections=False
+    ):
+        """
+        Function that adds host to host collection
+
+        Args:
+            entity_name: Name of the host
+            host_collection_name: Name of the host collection we want to remove host from
+            add_to_all_collections: If True, host will be added to all host collections
+
+        Raises:
+            ValueError: If specific hostColName is set to be removed and add all is set to True.
+            ValueError: If no parameters are passed.
+            ValueError: If host is already assigned to selected host collection.
+            ValueError: If there are no host collections left for addition.
+            ValueError: Given host collection name is not found in host collections.
+        """
+
+        if host_collection_name and add_to_all_collections:
+            raise ValueError(
+                'Either host_collection_name or add_to_all_collections must be provided!'
+            )
+        if (host_collection_name is None) and (add_to_all_collections is False):
+            raise ValueError(
+                'host_collection_name None and add_to_all_collections False '
+                'is invalid combination!'
+            )
+
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        # Handle the case where there are no host collections assigned
+        if view.overview.host_collections.no_host_collections.is_displayed:
+            view.overview.host_collections.add_to_host_collection.click()
+        else:
+            if (
+                host_collection_name
+                in view.overview.host_collections.assigned_host_collections.read()
+            ):
+                raise ValueError(f'{host_collection_name} already assigned to host!')
+            view.overview.host_collections.kebab_menu.item_select('Add host to collections')
+
+        view = ManageHostCollectionModal(self.browser)
+        self.browser.plugin.ensure_page_safe()
+        view.wait_displayed()
+
+        if view.create_host_collection.is_displayed:
+            raise ValueError('No host collections found or left for addition!')
+
+        if not add_to_all_collections:
+            view.searchbar.fill("name = " + host_collection_name, enter_timeout=5)
+            view.wait_displayed()
+            self.browser.plugin.ensure_page_safe()
+            if view.host_collection_table.row_count == 0:
+                raise ValueError(f'{host_collection_name} not found in host collections!')
+            time.sleep(3)
+            view.host_collection_table[0][0].widget.click()
```

Comment:
What is that? I suppose a checkbox? Either address better or if unable, comment.

Reply:
I will add comment